clevertap-core/src/main/java/com/clevertap/android/sdk/network/api/CtApi.kt

```diff
@@ -41,74 +46,192 @@ internal class CtApi(
     var currentRequestTimestampSeconds = 0
         private set
 
-    fun sendQueue(useSpikyDomain: Boolean, body: SendQueueRequestBody): Response =
-        httpClient.execute(createRequest(
-            relativePath = "a1",
-            body = body.toString(),
-            useSpikyDomain = useSpikyDomain,
-            includeTs = true)
+    fun sendQueue(
+        isViewedEvent: Boolean,
+        body: SendQueueRequestBody
+    ): Response =
+        httpClient.execute(
+            createRequest(
+                baseUrl = getActualDomain(isViewedEvent = isViewedEvent) ?: defaultDomain,
+                relativeUrl = "a1",
+                body = body.toString()
+            )
         )
 
-    fun performHandshakeForDomain(useSpikyDomain: Boolean): Response {
+    fun performHandshakeForDomain(isViewedEvent: Boolean): Response {
+        val baseUrl = getHandshakeDomain(isViewedEvent)
+
+        // append extra info in header in-case we are using custom handshake domain
+        val headers = if (customHandshakeDomain.isNotNullAndBlank() && baseUrl == customHandshakeDomain) {
+            defaultHeaders.plus(HEADER_CUSTOM_HANDSHAKE to customHandshakeDomain!!)
+        } else {
+            defaultHeaders
+        }
         val request = createRequest(
-            relativePath = "hello",
+            baseUrl = baseUrl,
+            relativeUrl = "hello",
             body = null,
-            useSpikyDomain = useSpikyDomain,
-            includeTs = false
+            includeTs = false,
+            headers = headers
         )
+
         logger.verbose(logTag, "Performing handshake with ${request.url}")
+
         return httpClient.execute(request)
     }
 
     fun defineVars(body: SendQueueRequestBody): Response =
         httpClient.execute(
-            createRequest("defineVars", body.toString(), useSpikyDomain = false, includeTs = true)
+            createRequest(
+                baseUrl = getActualDomain(isViewedEvent = false) ?: defaultDomain,
+                relativeUrl = "defineVars",
+                body = body.toString()
+            )
         )
 
     fun defineTemplates(body: DefineTemplatesRequestBody): Response =
         httpClient.execute(
-            createRequest("defineTemplates", body.toString(), useSpikyDomain = false, includeTs = true)
+            createRequest(
+                baseUrl = getActualDomain(isViewedEvent = false) ?: defaultDomain,
+                relativeUrl = "defineTemplates",
+                body = body.toString()
+            )
         )
 
-    fun getActualDomain(useSpikyDomain: Boolean): String? {
+    fun getActualDomain(isViewedEvent: Boolean): String? {
```

Comment:
Some refactoring 
```
fun getActualDomain(isViewedEvent: Boolean): String? {
    return when {
        !region.isNullOrBlank() -> buildDomain(region, isViewedEvent)
        !isViewedEvent && !proxyDomain.isNullOrBlank() -> proxyDomain
        isViewedEvent && !spikyProxyDomain.isNullOrBlank() -> spikyProxyDomain
        isViewedEvent -> cachedSpikyDomain
        else -> cachedDomain
    }
}

fun getHandshakeDomain(isViewedEvent: Boolean): String {

    return when {
        !region.isNullOrBlank() -> buildDomain(region, isViewedEvent)
        !isViewedEvent && !proxyDomain.isNullOrBlank() -> proxyDomain
        isViewedEvent && !spikyProxyDomain.isNullOrBlank() -> spikyProxyDomain      
        !customHandshakeDomain.isNullOrBlank() -> customHandshakeDomain
        isViewedEvent && !cachedSpikyDomain.isNullOrBlank() -> cachedSpikyDomain
        !isViewedEvent && !cachedDomain.isNullOrBlank() -> cachedDomain
        else -> defaultDomain
    }
}

// Helper function to build the domain string
private fun buildDomain(region: String, isViewedEvent: Boolean): String {
    return buildString {
        append(region)
        if (isViewedEvent) append(spikyRegionSuffix)
        append(".")
        append(defaultDomain)
    }
}
```

Reply:
when statement is unreadable hence kept it as if else with returns and made simpler.